application/common/middlewares.py

```diff
@@ -1,12 +1,13 @@
+from aws_lambda_powertools import Logger
 from aws_lambda_powertools.middleware_factory import lambda_handler_decorator
+from aws_lambda_powertools.utilities.data_classes import SQSEvent
 from aws_lambda_powertools.utilities.typing import LambdaContext
-from aws_lambda_powertools import Logger
 
 logger = Logger(child=True)
 
 
 @lambda_handler_decorator(trace_execution=True)
-def unhandled_exception_logging(handler, event, context):
+def unhandled_exception_logging(handler, event: SQSEvent, context: LambdaContext):
```

Comment:
would either leave this an untyped or Union all the possible Event types then this middleware can be used on any other future lambdas as doesnt really depend on anything specific in SQSEvent, although saying that all the lambdas in our proposed solution are SQS triggered